### ISSUE TITLE
Critical bug in web server!  exceptions.RuntimeError: cannot register two producers

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11453]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| slyphon reported|
|-|-|
|Trac ID|trac#348|
|Type|enhancement|
|Created|2003-10-17 11:39:47Z|

<details><summary>Searchable metadata</summary>

```
trac-id__348 348
type__enhancement enhancement
reporter__slyphon slyphon
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1066390787000000 1066390787000000
changetime__1468843437493842 1468843437493842
version__ 
owner__ 
cc__slyphon
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="slyphon's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|slyphon commented|
|-|-|

```
#!html
<pre>
2003/10/15 18:24 EDT [-] 192.168.1.50 - - [15/Oct/2003:22:24:48 +0000] "GET
/cgi-bin/man/man2html HTTP/1.1" 200 2383 "-" "Mozilla/5.0 (Windows; U; Windows
2003/10/15 18:25 EDT [HTTPChannel,0,192.168.1.50] Traceback (most recent call last):
      File "/usr/lib/python2.3/site-packages/twisted/web/server.py", line 166,
in render
        body = resrc.render(self)
      File "/usr/lib/python2.3/site-packages/twisted/web/twcgi.py", line 123, in
render
        self.runProcess(env, request, qargs)
      File "/usr/lib/python2.3/site-packages/twisted/web/twcgi.py", line 128, in
runProcess
        reactor.spawnProcess(p, self.filename, [self.filename]+qargs, env,
os.path.dirname(self.filename))
      File "/usr/lib/python2.3/site-packages/twisted/internet/default.py", line
158, in spawnProcess
        return process.Process(self, executable, args, env, path,
processProtocol, uid, gid)
    --- &lt;exception caught here> ---
      File "/usr/lib/python2.3/site-packages/twisted/internet/process.py", line
283, in __init__
        self.proto.makeConnection(self)
      File "/usr/lib/python2.3/site-packages/twisted/internet/protocol.py", line
305, in makeConnection
        self.connectionMade()
      File "/usr/lib/python2.3/site-packages/twisted/web/twcgi.py", line 193, in
connectionMade
        self.request.registerProducer(self, 1)
      File "/usr/lib/python2.3/site-packages/twisted/protocols/http.py", line
570, in registerProducer
        self.transport.registerProducer(producer, streaming)
      File "/usr/lib/python2.3/site-packages/twisted/internet/abstract.py", line
234, in registerProducer
        raise RuntimeError("cannot register two producers")
    exceptions.RuntimeError: cannot register two producers
    
this happens every time i browse to my man2html cgi page. the page still
functions properly. 

Glyph said this is Very Bad Thing.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Duplicate of 345.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset dd8df60e30c752d62d2c86b25be104928a78e408
```
#!CommitTicketReference repository="" revision="dd8df60e30c752d62d2c86b25be104928a78e408"
Merge pull request #348 from twisted/8619-rodrigc-spread

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8619

In twisted/spread, fix use of list and range
```
